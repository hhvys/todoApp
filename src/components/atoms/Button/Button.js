import React from 'react';
import './Button.css';

const Button = ({text, className, ...props}) => (
	<div className={`div__button ${className ? className : ''}`} {...props}>
		{text}
	</div>
);

export default Button;