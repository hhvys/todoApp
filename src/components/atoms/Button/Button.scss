.button {
    background: rgba(102,137,100,0.75);
    border-radius: 5px;
    border: none;
    color: white;
    padding: 3px 10px;
    font-weight: 400;
    font-size: 11px;
    width: fit-content;
    max-width: 100%;
    cursor: pointer;
    height: 22px;
}
