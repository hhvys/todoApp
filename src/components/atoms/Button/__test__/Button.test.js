import React from 'react';
import Button from '../Button';

describe('component', () => {
	describe('Button', () => {
		it('should render', () => {
			expect(toJSON(shallow(<Button/>))).toMatchSnapshot();
		});
		it('should render className', () => {
			const props = {
				className: 'className'
			};
			expect(toJSON(shallow(<Button {...props}/>))).toMatchSnapshot();
		});
		it('should render textContent', () => {
			const props = {
				text: 'text'
			};
			expect(toJSON(shallow(<Button {...props}/>))).toMatchSnapshot();
		});
	});
});
