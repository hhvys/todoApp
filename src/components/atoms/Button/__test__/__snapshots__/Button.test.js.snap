// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Button should render 1`] = `
<div
  className="button "
/>
`;

exports[`component Button should render className 1`] = `
<div
  className="button className"
/>
`;

exports[`component Button should render textContent 1`] = `
<div
  className="button "
>
  text
</div>
`;
